st2reactor/st2reactor/container/sensor_wrapper.py

```diff
@@ -67,6 +75,120 @@ def dispatch(self, trigger, payload=None):
 
+    def get_value(self, name):
+        """
+        Retrieve a value from the datastore for the provided key.
+
+        :param name: Key name.
+        :type name: ``str``
+
+        :rtype: ``str`` or ``None``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        self._logger.audit('Retrieving value from the datastore (name=%s)', name)
+
+        try:
+            kvp = client.keys.get_by_id(id=name)
+        except Exception:
+            return None
+
+        if kvp:
+            return kvp.value
+
+        return None
+
+    def set_value(self, name, value, ttl=None):
+        """
+        Set a value for the provided key.
+
+        :param name: Key name.
+        :type name: ``str``
+
+        :param value: Key value.
+        :type value: ``str``
+
+        :param ttl: Optional TTL (in seconds).
+        :type ttl: ``int``
+
+        :return: ``True`` on sucess, ``False`` otherwise.
+        :rtype: ``bool``
+        """
+        value = str(value)
+
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        self._logger.audit('Setting value in the datastore (name=%s)', name)
+
+        instance = KeyValuePair()
+        instance.id = name
+        instance.name = name
+        instance.value = value
+
+        if ttl:
+            instance.ttl = ttl
+
+        client.keys.update(instance=instance)
+        return True
+
+    def delete_value(self, name):
+        """
+        Delete the provided key.
+
+        :param name: Name of the key to delete.
+        :type name: ``str``
+
+        :return: ``True`` on sucess, ``False`` otherwise.
+        :rtype: ``bool``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        instance = KeyValuePair()
+        instance.id = name
+        instance.name = name
+
+        self._logger.audit('Deleting value from the datastore (name=%s)', name)
+
+        try:
+            client.keys.delete(instance=instance)
+        except Exception:
+            return False
+
+        return True
+
+    def _get_api_client(self):
+        """
+        Retrieve API client instance.
+        """
+        # TODO: API client is really unfriendly and needs to be re-designed and
+        # improved
+        api_url = os.environ.get(API_URL_ENV_VARIABLE_NAME, None)
+        auth_token = os.environ.get(AUTH_TOKEN_ENV_VARIABLE_NAME, None)
+
+        if not api_url or not auth_token:
+            raise ValueError('%s and %s environment variable must be set' %
+                             (API_URL_ENV_VARIABLE_NAME, AUTH_TOKEN_ENV_VARIABLE_NAME))
+
+        if not self._client:
+            self._client = Client(api_url=api_url)
+
+        return self._client
+
+    def _get_full_key_name(self, name):
+        """
+        :rtype: ``str``
+        """
+        prefix = self._get_datastore_key_prefix()
```

Comment:
Agreed, but that's more of a feature and not a bug :)

If you rename a pack, this usually means things changed substantially / whatever so not being able to access old / existing keys is fine imo.

And that's also one of the reasons where TTL comes handy. Besides TTL, I've also added `st2  key delete_by_prefix <key prefix>` command which makes it easy to delete all the keys belonging to a particular sensor.